zookeeper-server/src/main/java/org/apache/zookeeper/server/DataTree.java

```diff
@@ -964,7 +974,9 @@ public ProcessTxnResult processTxn(TxnHeader header, Record txn, boolean isSubTx
 
                         ByteBufferInputStream.byteBuffer2Record(bb, record);
 
-                        if (failed && subtxn.getType() != OpCode.error){
+                        if (failed && subtxn.getType() != OpCode.error &&
+                            // Before the error, the return values of getter methods are valid.
```

Comment:
Please add more details to the comments, because it's hard to understand the intention here. Something like:
> For `getChildren` requests we'd like to the see the results of requests which was made before the error in the multi.